### ISSUE TITLE
aapt2 + mobile-install v1 doesn't build a runnable APK

### ISSUE THREAD

**Issue by jin** [bazelbuild/bazel#5799]
Bug report from @sunyal:

Building the example app from [Bazel Android Tutorial ](https://docs.bazel.build/versions/master/tutorial/android-app.html#build-the-app) with `--android_aapt=aapt2` results in a un-runnable app. App is fine with `--android_aapt=aapt`.

```
    --------- beginning of crash
08-06 16:37:41.616 11490-11490/com.google.bazel.example.android E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.google.bazel.example.android, PID: 11490
    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.google.bazel.example.android/com.google.bazel.example.android.activities.MainActivity}: android.content.res.Resources$NotFoundException: Resource ID #0x7f040000 type #0x5 is not valid
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2778)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2856)
        at android.app.ActivityThread.-wrap11(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1589)
```

**Comment by jin** [comment]
Reproduced:

```
08-08 11:42:21.568  6551  6551 D AndroidRuntime: Shutting down VM
08-08 11:42:21.569  6551  6551 E AndroidRuntime: FATAL EXCEPTION: main
08-08 11:42:21.569  6551  6551 E AndroidRuntime: Process: com.google.bazel.example.android, PID: 6551
08-08 11:42:21.569  6551  6551 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{com.google.bazel.example.android/com.google.bazel.example.android.activities.MainActivity}: android.content.res.Resources$NotFoundException: Resource ID #0x7f040000 type #0x5 is not valid
08-08 11:42:21.569  6551  6551 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2817)
08-08 11:42:21.569  6551  6551 E AndroidRuntime: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2892)
08-08 11:42:21.569  6551  6551 E AndroidRuntime: 	at android.app.ActivityThread.-wrap11(Unknown Source:0)
08-08 11:42:21.569  6551  6551 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1593)
08-08 11:42:21.569  6551  6551 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:105)
08-08 11:42:21.569  6551  6551 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:164)
08-08 11:42:21.569  6551  6551 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6541)
08-08 11:42:21.569  6551  6551 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
08-08 11:42:21.569  6551  6551 E AndroidRuntime: 	at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
08-08 11:42:21.569  6551  6551 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
08-08 11:42:21.569  6551  6551 E AndroidRuntime: Caused by: android.content.res.Resources$NotFoundException: Resource ID #0x7f040000 type #0x5 is not valid
08-08 11:42:21.569  6551  6551 E AndroidRuntime: 	at android.content.res.Resources.loadXmlResourceParser(Resources.java:2138)
08-08 11:42:21.569  6551  6551 E AndroidRuntime: 	at android.content.res.Resources.getLayout(Resources.java:1142)
08-08 11:42:21.569  6551  6551 E AndroidRuntime: 	at android.view.LayoutInflater.inflate(LayoutInflater.java:421)
08-08 11:42:21.569  6551  6551 E AndroidRuntime: 	at android.view.LayoutInflater.inflate(LayoutInflater.java:374)
08-08 11:42:21.569  6551  6551 E AndroidRuntime: 	at com.android.internal.policy.PhoneWindow.setContentView(PhoneWindow.java:418)
08-08 11:42:21.569  6551  6551 E AndroidRuntime: 	at android.app.Activity.setContentView(Activity.java:2654)
08-08 11:42:21.569  6551  6551 E AndroidRuntime: 	at com.google.bazel.example.android.activities.MainActivity.onCreate(MainActivity.java:42)
08-08 11:42:21.569  6551  6551 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:6975)
08-08 11:42:21.569  6551  6551 E AndroidRuntime: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1213)
08-08 11:42:21.569  6551  6551 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2770)
08-08 11:42:21.569  6551  6551 E AndroidRuntime: 	... 9 more
```

**Comment by jin** [comment]
@sunyal I can reproduce this with `bazel mobile-install` but not `bazel build && adb install`. Did you try it with `mobile-install`?

**Comment by sunyal** [comment]
@jingwen I believe the command that I was using was `mobile-install`

**Comment by sunyal** [comment]
@jin if I recall it was through `bazel mobile-install`

**Comment by sunyal** [comment]
@jin does it work if you call `bazel build` and then use the `adb install` or are you also able to use the `build mobile-install` after it is built?

**Comment by sunyal** [comment]
@jin I was able to install it using the `adb install` command after just issuing the `bazel build`. It looks like the issue is related to `bazel mobile-install`

**Comment by jin** [comment]
@sunyal thanks for confirming! It's definitely an issue with mobile-install.

**Comment by jin** [comment]
Hey @sunyal, the `mobile-install` built in Bazel is actually v1 of the command. There is `mobile-install` v2, a rewrite of v1, due to be open sourced this quarter. That version is more feature packed and works with aapt2. Since v2 will be replacing v1, I'm downgrading this to P3 as v1 will eventually go away. 

For now, the workaround to use aapt2 would be to use `bazel build` and `adb install`. Sorry for the inconvenience.

**Comment by sunyal** [comment]
@jin thanks for the update

**Comment by jin** [comment]
@sunyal this is fixed, and will be available in 0.22 (or head now)
